src/runtime/tuple.c

```diff
@@ -185,6 +187,32 @@ tuple allocate_tuple(void)
     return tag(allocate_table(theap, key_from_symbol, pointer_equal), tag_table_tuple);
 }
 
+closure_function(1, 2, boolean, clone_tuple_each,
+                 tuple, clone,
+                 value, s, value, v)
+{
+    if (is_tuple(v))
+        v = clone_tuple(v);
+    else
```

Comment:
This needs to be updated handle the new tag types.

Reply:
OK, the clone_tuple_each closure now checks whether a value is immediate (and also whether it's `null_value`), and handles the value based on its tag.